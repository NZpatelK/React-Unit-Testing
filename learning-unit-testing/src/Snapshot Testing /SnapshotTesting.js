import React from 'react'

const SnapshotTesting = ({text}) => {
  return (
    <div data-testid= "text">
        <h1>{text}</h1>
    </div>
  )
}

export default SnapshotTesting