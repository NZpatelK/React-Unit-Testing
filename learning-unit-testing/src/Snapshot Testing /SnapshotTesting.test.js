import { render, screen, cleanup } from "@testing-library/react";
import * as snapshot from "react-test-renderer";
import SnapshotTesting from "./SnapshotTesting";
import "@testing-library/jest-dom";

/**
 * Snapshot is a useful tool for unit testing React components 
 * by capturing and comparing their rendered output.
 */

test("text matches snapshot", () => {
   const component = snapshot.create(<SnapshotTesting/>); 
   let tree = component.toJSON();
   expect(tree).toMatchSnapshot();


  });