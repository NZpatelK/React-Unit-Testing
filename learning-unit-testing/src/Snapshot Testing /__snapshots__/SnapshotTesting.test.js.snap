// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text matches snapshot 1`] = `
<div
  data-testid="text"
>
  <h1 />
</div>
`;
